gossip-base/src/main/java/org/apache/gossip/manager/GossipCore.java

```diff
@@ -111,11 +120,23 @@ public void addPerNodeData(PerNodeDataMessage message){
       PerNodeDataMessage current = nodeMap.get(message.getKey());
       if (current == null){
         nodeMap.putIfAbsent(message.getKey(), message);
+        notifyPerNodeData(message,null);
       } else {
         if (current.getTimestamp() < message.getTimestamp()){
           nodeMap.replace(message.getKey(), current, message);
+          notifyPerNodeData(message,current.getPayload());
         }
       }
+    } else {
+      notifyPerNodeData(message,null);
```

Comment:
Make sure to test with CRDT types we may need to check for equality of the objects so that we do not even with new objects that are logically equal